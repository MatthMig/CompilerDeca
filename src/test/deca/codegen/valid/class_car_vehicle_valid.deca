//  Description : 
// 
//      Test correct qui utilise deux classes, vehicule et voiture. La classe voiture hérite de la classe véhicule.
// 
//  Résultats :
// 
//      N/A 
// 
//  Historique : 
// 
//      Créé le 13/06 par gl03

class vehicule {

    protected int nbRoues, nbPortes , chevaux , km;
    boolean reglementaire;
    int consommation;
    boolean vendu;
    void casse() {
        println("Le vehicule", this ,"est à la casse");
        this=null;
    }
    void seDeplacer(int distance) {
    if(this.reglementaire==true){
            this.km=this.km+distance;
            println("arrivée à destination");
    }
    else {
        println("voyage impossible");
    }
    
}

}

class voiture extends vehicule { 


    void casseVoiture(int finDeVie){
        if(this.km<=finDeVie){
            this.reglementaire=true;
            println("la voiture est en état de rouler");
        }
        else {
            this.reglementaire=false;
            println("la voiture est bonne pour la casse");
            this.casse();
        }
    }

    void venduVoiture(){
        this.vendu=true;
    }
}

{
    voiture rs6SlineBreakBiTurbo = new voiture();
    vehicule trotinetteNulle = new vehicule();
    rs6SlineBreakBiTurbo.nbRoues=4;
    rs6SlineBreakBiTurbo.nbPortes=5;
    rs6SlineBreakBiTurbo.chevaux=630;
    rs6SlineBreakBiTurbo.km=10000;
    rs6SlineBreakBiTurbo.reglementaire=true;
    rs6SlineBreakBiTurbo.vendu=false;

    
    trotinetteNulle.nbRoues=2;
    trotinetteNulle.nbPortes=0;
    trotinetteNulle.chevaux=1;
    trotinetteNulle.km=50;
    trotinetteNulle.reglementaire=true;
    trotinetteNulle.vendu=false;

    rs6SlineBreakBiTurbo.seDeplacer(1000);
    rs6SlineBreakBiTurbo.casseVoiture(15000);
    trotinetteNulle.seDeplacer(3);
    trotinetteNulle.casse();
    


}

