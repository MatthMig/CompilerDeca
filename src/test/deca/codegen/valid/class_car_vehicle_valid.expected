rs6SlineBreakBiTurbo arrivé à destination
la voiture est bonne pour la casse
le vehicule rs6SlineBreakBiTurbo est à la casse 
trotinetteNulle arrivé à destination 
le vehicule trotinetteNulle est à la casse